### ISSUE TITLE
XNLI-15 vs one language pair alignment

### ISSUE THREAD

**Issue by dominikabasaj** [facebookresearch/XLM#84]
Hi!

I really enjoyed reading the paper. I would like to ask for some additional clarification regarding the model with TLM objective. 

In the paper, as an example you use English-French language pair, however, the released model with TLM objective is applicable to 15 languages.

1. Does it mean that the model was trained on the corpus of parallel sentences of these 15 languages?
2. Taking that into consideration is it possible to somehow ensure that English sentence would use only French context or does it require training of TLM only on English-French pairs? 

My question refers to this citation from your paper:

> To predict a word masked in an English sentence, the model can either attend to surrounding English words or to the French translation, encouraging the model to align the English and French representations. In particular, the model can leverage the French context if the English one is not sufficient to infer the masked English words.
> 

Thank you!

**Comment by glample** [comment]
Hi,

For English-French, we never used TLM. Released En-Fr, En-DE and En-Ro models only use MLM and are not trained with parallel data. TLM was used only for the XNLI model. For the 15 languages, we have 15 corpus (each En-XX for XX any of the 14 other languages) and we only train with the TLM step for pairs of sentences where one language is English.

**Comment by dominikabasaj** [comment]
Ok, thank you understood. 

So, if I am interested only in Eng-French embedding alignment than it would be more beneficial to train the model with TLM objective from scratch for this particular language pair, right?

**Comment by glample** [comment]
Yes. Or you could finetune the current model En-Fr that was trained with MLM only.

**Comment by dominikabasaj** [comment]
Thank you!
